core/model/src/payment.rs

```diff
@@ -69,37 +70,62 @@ pub mod local {
     }
 
     #[derive(Clone, Debug, Serialize, Deserialize)]
-    pub struct Init {
-        pub identity: NodeId,
-        pub provider: bool,
-        pub requestor: bool,
+    pub struct RegisterAccount {
+        pub driver: String,
+        pub platform: String,
+        pub address: String,
+        pub mode: AccountMode,
     }
 
-    impl RpcMessage for Init {
-        const ID: &'static str = "init";
+    #[derive(Clone, Debug, Serialize, Deserialize, thiserror::Error)]
+    pub enum RegisterAccountError {
+        #[error("Account already registered")]
+        AlreadyRegistered,
+        #[error("Error while registering account: {0}")]
+        Other(String),
+    }
+
+    impl RpcMessage for RegisterAccount {
+        const ID: &'static str = "RegisterAccount";
         type Item = ();
-        type Error = GenericError;
+        type Error = RegisterAccountError;
     }
 
     #[derive(Clone, Debug, Serialize, Deserialize)]
-    pub struct GetStatus(NodeId);
+    pub struct UnregisterAccount {
+        pub platform: String,
```

Comment:
IMHO `driver` should be better and consistent with `RegisterAccount`

Reply:
The mapping stored by payment service is `(platform, address) -> (driver, mode)`. To remove an entry from a map we need to give the key as parameter. `driver` is not part of the key, while `platform` is.